flink-cdc-common/src/main/java/org/apache/flink/cdc/common/factories/DataSourceFactory.java

```diff
@@ -28,4 +28,7 @@ public interface DataSourceFactory extends Factory {
 
     /** Creates a {@link DataSource} instance. */
     DataSource createDataSource(Context context);
+
+    /** Checking if this {@link DataSource} could be created in batch mode. */
+    default void verifyRuntimeMode(Context context) {}
```

Comment:
This is not intuitive to me, why do we need to introduce a method to verify runtime mode(especially the java doc said batch mode) in Factory?

Reply:
Hi. @leonardBang The purpose of adding this method is to remind developers to handle streaming and batch processing when developing new data sources. It's also feasible to remove the definition of this method and let developers implement it themselves in the createDataSource method. So, should we remove it now? I'm glad to hear your opinion.